Sources/MapboxNavigation/NavigationMapView.swift

```diff
@@ -1429,6 +1453,8 @@ open class NavigationMapView: UIView {
                 symbolLayer.textOpacity = .constant(0.75)
                 symbolLayer.textAnchor = .constant(.bottom)
                 symbolLayer.textJustify = .constant(.left)
+                symbolLayer = delegate?.navigationMapView(self, willAdd: symbolLayer) as? SymbolLayer ?? symbolLayer
```

Comment:
I think that it makes sense to follow design by contract approach and assert whenever user provides type that is not acceptable (e.g. by using `preconditionFailure()`). Let's do that in all other places as well.

Reply:
Thanks for the suggestion, the layer type check is added with the `preconditionFailure`.